src/LearnositySdk/Utils/Json.php

```diff
@@ -104,4 +111,67 @@ public static function isJson($val)
     {
         return !is_null(json_decode($val, true));
     }
+
+    /**
+     * This function builds up a map of float values in scientific notation and the equivalent float
+     * notations. json_encode encodes float values in scientific notation, but we want to make sure
+     * they are in float notation
+     *
+     * @param $json the $json to be encoded
+     * @param $floatMap pointer to the floatMap
+     */
+    private static function getHighPrecisionFloatMap($json, &$floatMap)
+    {
+        if (is_array($json)) {
+            foreach ($json as $value) {
+                if (is_array($value)) {
+                    self::getHighPrecisionFloatMap($value, $floatMap);
+                }
+                if (is_float($value)) {
+                    $currentMap = self::getFloatMap($value);
+                    if ($currentMap) {
+                        $floatMap[] = $currentMap;
+                    }
+                }
+            }
+        } elseif (is_float($json)) {
+            $currentMap = self::getFloatMap($json);
+            if ($currentMap) {
+                $floatMap[] = $currentMap;
+            }
+        }
+    }
+
+    /**
+     * Helper function to build up the float map
+     *
+     * @param $value
+     * @return array|bool
+     */
+    private static function getFloatMap($value)
+    {
+        $stringValue = json_encode($value);
+
+        // If the string value is in scientific notation it should have a negative exponent. Split it there.
+        $explodedValue = explode('E-', strtoupper($stringValue));
+        if (count($explodedValue) === 2) {
+            // The second entry tells us how many decimal places are created via the exponent
+            $decimalsFromExponent = (int)$explodedValue[1];
+
+            // The base hase the form x.xxxxx where x is any digit. We need to get the number of decimals after
+            // the dot
+            $decimalsFromBase = 0;
+            $explodedBase = explode('.', $explodedValue[0]);
+            if (count($explodedBase) === 2) {
+                $decimalsFromBase = strlen($explodedBase[1]);
+            }
+            $totalDecimals = $decimalsFromBase + $decimalsFromExponent;
+            $stringValueAsFloat = number_format($value, $totalDecimals);
+            return [
+                'scientificValue' => $stringValue,
+                'floatValue' => $stringValueAsFloat
+            ];
+        }
+        return false;
```

Comment:
Returning either an array or a boolean is a bit weird here. Could we just return an empty array here, and check for `!empty($currentMap)` in the code using it?